From d70313d6db15fdbb5b0852f18ac746e2dd70e8f0 Mon Sep 17 00:00:00 2001
From: John Williams <john.williams@petalogix.com>
Date: Thu, 27 May 2010 16:33:13 +1000
Subject: [PATCH] powerpc: auto-calculate link_address of boot wrapper to fit any kernel image

Fixing the link_adddress to 0x400000 fails when a large initramfs is linked.

A little bit of awk-foo to set the link address to the next megabyte boundary
is a simple, generic fix.

Signed-off-by: John Williams <john.williams@petalogix.com>
---
 arch/powerpc/boot/wrapper |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/boot/wrapper b/arch/powerpc/boot/wrapper
index f6cf4b0..9634878 100755
--- a/arch/powerpc/boot/wrapper
+++ b/arch/powerpc/boot/wrapper
@@ -143,9 +143,12 @@ objflags=-S
 tmp=$tmpdir/zImage.$$.o
 ksection=.kernel:vmlinux.strip
 isection=.kernel:initrd
-link_address='0x400000'
 make_space=y
 
+# default auto-calculate link_address to make room for the kernel
+# round up kernel image size to nearest megabyte
+link_address=`${CROSS}size -x ${kernel} | grep ${kernel} | awk '{printf("0x%08x", and($4 + 0x0fffff, 0xfffe0000))}'`
+
 case "$platform" in
 pseries)
     platformo=$object/of.o
-- 
1.7.0.4

